api/src/test/java/org/hawkular/apm/api/utils/EndpointUtilTest.java

```diff
@@ -96,6 +96,8 @@ public void testDecodeNonClientURI() {
     @Test
     public void testGetSourceEndpointConsumer() {
         Trace trace = new Trace();
+        trace.setTraceId("1");
```

Comment:
I see otherwise it would fail on NPE `Trace#getinitialFragment()` :+1: . This reminds me if it is mentioned gitbook#writing OT provider that initial fragment should have the same id as the trace.